modules/navier_stokes/test/tests/postprocessors/tests

```diff
@@ -0,0 +1,86 @@
+[Tests]
+  design = 'VolumetricFlowRate.md InternalVolumetricFlowRate.md'
+  issues = '#16169 #16585'
+  [fe]
+    type = CSVDiff
+    input = conservation_INSFE.i
+    csvdiff = straight_INSFE.csv
+    abs_zero = 1e-8
+    requirement = 'The system shall be able to compute mass and momentum flow rates at internal and external boundaries of a straight channel with a finite element incompressible Navier Stokes model.'
+    cli_args = "Mesh/inactive='diverging_mesh' Outputs/file_base='straight_INSFE'"
+  []
+  [fe_diverging]
+    type = CSVDiff
+    input = conservation_INSFE.i
+    csvdiff =  diverging_INSFE.csv
+    abs_zero = 1e-8
+    requirement = 'The system shall be able to compute mass and momentum flow rates at internal and external boundaries of a diverging channel with a finite element incompressible Navier Stokes model.'
+    cli_args = "Outputs/file_base='diverging_INSFE'"
+  []
+  [fv_straight]
+    type = CSVDiff
+    input = conservation_INSFV.i
+    csvdiff =  straight_INSFV.csv
+    abs_zero = 1e-8
+    requirement = 'The system shall be able to compute flow rates and prove mass, momentum and energy conservation at internal and external boundaries of a frictionless heated straight channel with a finite volume incompressible Navier Stokes model.'
+    cli_args = "Mesh/inactive='diverging_mesh' Outputs/file_base='straight_INSFV'"
+    ad_indexing_type = 'global'
+  []
+  [fv_diverging]
+    requirement = 'The system shall be able to compute flow rates and prove mass, momentum and energy conservation at internal and external boundaries of a frictionless heated diverging channel with a finite volume incompressible Navier Stokes model,'
+    [fv_quad_xy]
+      type = CSVDiff
+      input = conservation_INSFV.i
+      csvdiff = diverging_INSFV_quad_xy.csv
+      abs_zero = 1e-8
+      detail = 'with a quadrilateral mesh in XY geometry, with mass flow measured using either a variable or material property.'
```

Comment:
for these details, I believe you want a comma instead of a period. Other than that these are fantastic requirements!